src/styles/container.scss

```diff
@@ -0,0 +1,47 @@
+.container {
+  display: grid;
+  width: 100%;
+
+  &__item {
+    display: grid;
+    justify-content: center;
+    align-items: center;
+    height: $itemHeight;
+    color: #fff;
+
+    @for $num from 1 to 7 {
+      &--#{$num}::before {
+        content: "#{$num}";
+      }
+    }
+
+    @each $num, $color in $colors {
+      &--#{$num} {
+        background-color: #{$color};
+      }
+    }
+  }
+
+  @media (min-width: 300px) {
+    & {
+      grid-template-columns: 1fr;
+    }
+  }
+
+  @media (min-width: 600px) {
+    & {
+      grid-template-columns: repeat(2, 1fr);
+
+    }
+
+    &__item--1 {
+      grid-column-start: 1;
+    }
+  }
+
+  @media (min-width: 900px) {
```

Comment:
You don't need to write three @media in your code, it's possible to solve this task only with two @media by setting min-width for your items, then defining a usual behavior for your container and then defining how your container should behave when the max-width: 899 and 599 px.